### ISSUE TITLE
Issue with config file paths for plugins in Godot 3.2.4.rc3

### ISSUE THREAD

**Issue by coppolaemilio** [godotengine/godot#46383]
**Godot version:**
Godot 3.2.4.rc3

**OS/device including version:**
Windows 10

**Issue description:**
I'm developing a plugin and also using one by Will Nations (https://github.com/godot-extended-libraries/godot-plugin-refresher) to refresh the plugin when I make big changes to it. Since 3.2.4.rc3 I'm experiencing some minor issues (can't reproduce at the moment all of them) but Will was able to identify the possible origin of one of them: 

> The problem appears to be (to me) that an attempt was made to simplify the logic related to identifying what a plugin inherently was (shifting away from a dependency on the .cfg file), but in the process they messed up the logic for identifying the existence of the addon. You can see in 3.2-stable, here where they fetch the config file and get its base dir:

> res://addons/godot-plugin-refresher/plugin.cfg -> res://adons/godot-plugin-refresher

> ...but then in the latest 3.2 here, they remove the config file path and use just the addon's subdirectory and try to get its base dir:

> godot-plugin-refresher -> ""

> This then results in the addon never being found in any conceivable case, afaict. If that isn't the issue, then I'm not sure what it is since, of all methods among the 7 mentions of the plugin_addons data structure throughout the EditorNode, that is the only line of code that is any different between the two versions of the engine.

(Issue in the plugin repo: https://github.com/godot-extended-libraries/godot-plugin-refresher/issues/13)

I'm not sure if this is the reason why the error is happening but I can't seem to find a solution.

**Steps to reproduce:**
Open the editor, try to reload the plugin pressing on the refresh arrow
![image](https://user-images.githubusercontent.com/2206700/109016009-02523700-76b6-11eb-805e-ec06889174fd.png)
You will see the error:
` editor/editor_node.cpp:3241 - Addon 'explore-editor-theme' failed to load. No directory found. Removing from enabled plugins`

**Minimal reproduction project:**
[MinimalPluginError.zip](https://github.com/godotengine/godot/files/6036682/MinimalPluginError.zip)

**Comment by akien-mga** [comment]
Likely a consequence of #45316, CC @Shatur95.

**Comment by pycbouh** [comment]
Yes, it's a regression from #45316. Calls to `get_editor_interface().is_plugin_enabled` are broken because now the entire path is stored internally, but this method expect the user to pass only the name.

```gdscript
print("explore-editor-theme -> ", get_editor_interface().is_plugin_enabled("explore-editor-theme"))
print("res://addons/explore-editor-theme -> ", get_editor_interface().is_plugin_enabled("res://addons/explore-editor-theme"))
print("res://addons/explore-editor-theme/plugin.cfg -> ", get_editor_interface().is_plugin_enabled("res://addons/explore-editor-theme/plugin.cfg"))
```

```
explore-editor-theme -> False
res://addons/explore-editor-theme -> False
res://addons/explore-editor-theme/plugin.cfg -> True
```

**Comment by Shatur95** [comment]
@pycbouh, @coppolaemilio, could you check #46389?
Sorry for the inconvenience.

**Comment by coppolaemilio** [comment]
I'll try to compile and check if it keeps happening. Never done it before so wish me luck 😬

**Comment by coppolaemilio** [comment]
Fixed on my end ✅

**Comment by akien-mga** [comment]
Fixed by #46389.
